src/app/components/MediaLoader/configs/portraitClipMedia.ts

```diff
@@ -8,92 +7,110 @@ import {
   adsEnabled = false,
   showAdsBasedOnLocation = false,
 }: ConfigBuilderProps): ConfigBuilderReturnProps => {
-  const portraitClipMediaBlock: PortraitClipMediaBlock = filterForBlockType(
+  const firstBlock = filterForBlockType(
     blocks,
     'portraitClipMedia',
-  );
-
-  const { images, video, type } = portraitClipMediaBlock?.model || {};
-
-  const { source, urlTemplate: locator } = images?.[1] ?? {};
-
-  const originCode = source?.replace('Image', '');
-
-  const versionID = video?.version?.id;
-
-  const clipISO8601Duration = video?.version?.duration;
-
-  const rawDuration = moment.duration(clipISO8601Duration).asSeconds();
-
-  const title = video?.title;
-
-  const videoId = video?.id;
-
-  const captionBlock = getCaptionBlock(blocks, 'live');
-
-  const caption =
-    captionBlock?.model?.blocks?.[0]?.model?.blocks?.[0]?.model?.text;
-
-  const kind = video?.version?.kind || 'programme';
-
-  const guidanceMessage = video?.version?.guidance;
+  ) as PortraitClipMediaBlock;
+
+  if (!firstBlock) {
+    return {
+      mediaType: 'video',
+      playerConfig: basePlayerConfig,
+      showAds: false,
+      orientation: 'portrait',
+    };
+  }
+
+  const portraitClipMediaBlocks: PortraitClipMediaBlock[] = filterForBlockType(
+    blocks,
+    'portraitClipMedia',
+    { multiple: true },
+  ) as PortraitClipMediaBlock[];
+
+  const playlistItems: PlaylistItem[] = portraitClipMediaBlocks.map(block => {
+    const { model } = block;
+    const { video, images } = model;
+    const version = video?.version;
+
+    // prefer portrait-oriented image if available (based on our bff structure) fallback to first available
+    const image = images?.[1] || images?.[0];
+    const originCode = image?.source?.replace('Image', '');
+    const locator = image?.urlTemplate;
+
+    return {
+      versionID: version?.id,
+      kind: version?.kind || 'programme',
+      duration: moment.duration(version?.duration || 'PT0S').asSeconds(),
+      embedRights: video?.isEmbeddingAllowed ? 'allowed' : undefined,
+      vpid: video?.id,
+      serviceID: version?.territories?.[0],
+      title: video?.title,
+      guidance: version?.guidance,
+      territories: version?.territories,
+      images,
+      holdingImageURL:
+        originCode && locator
+          ? buildIChefURL({ originCode, locator, resolution: DEFAULT_WIDTH })
+          : undefined,
+    };
+  });
 
   const showAds = shouldDisplayAds({
     adsEnabled,
     showAdsBasedOnLocation,
-    duration: rawDuration,
+    duration: playlistItems[0]?.duration ?? 0,
   });
 
-  const embeddingAllowed = video?.isEmbeddingAllowed ?? false;
+  if (showAds) {
+    playlistItems.unshift({ kind: 'advert' });
+  }
 
   const holdingImageURL = buildIChefURL({
-    originCode,
-    locator,
+    originCode:
+      firstBlock?.model?.images?.[1]?.source?.replace('Image', '') ?? '',
+    locator: firstBlock?.model?.images?.[1]?.urlTemplate ?? '',
     resolution: DEFAULT_WIDTH,
   });
```

Comment:
The URL this generates isn't quite right here or in the playlist map.

The `images` array should have a fully qualified name I think, so it should be pre-constructed with maybe just the `{width}` needing manually set.

Reply:
so does that mean we can just do 

`const holdingImageId = model?.video?.holdingImage?.id;` for the url that's stored within the `holdingImage` object on the BFF and then do this for now (and inside the map):

```
      holdingImageURL: holdingImageId
        ? buildIChefURL({
            originCode: 'pips',
            locator: holdingImageId,
            resolution: DEFAULT_WIDTH,
          })
        : undefined,
    };
```

the `url` is mostly formed it just needs `$recipe` for `buildIChefUrl` i think. `OriginCode` is hardcoded in there